### ISSUE TITLE
Metrics/MethodLength heredoc and method_call not completely compatible with each other?

### ISSUE THREAD

**Issue by hacktivista** [rubocop/rubocop#12001]
## Expected behavior

Should count heredoc lines as one when using the following configuration:

```yaml
Metrics/MethodLength:
  CountAsOne: ['heredoc', 'method_call']
```

## Actual behavior

With some specific conditions, it counts each line in heredoc within a method.

Conditions are: Don't use parentheses on a method call that uses heredoc and call something else after that method call
 
```sh
For /home/x/test_proj: configuration from /home/x/test_proj/.rubocop.yml
Default configuration from /home/x/.local/share/gem/ruby/3.2.0/gems/rubocop-1.53.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/x/test_proj/test.rb
C

Offenses:

test.rb:5:3: C: Metrics/MethodLength: Method has too many lines. [11/10]
  def test_method ...
  ^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
Finished in 0.2313520830066409 seconds
```

## Steps to reproduce the problem

Configure `rubocop.yml`

```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.1

Metrics/MethodLength:
  CountAsOne: ['heredoc', 'method_call']
```

Create a file `test.rb`

```
# frozen_string_literal: true

# Test class
class Test
  def test_method
    other_method <<~ANY
      2
      3
      4
      5
      6
      7
      8
      9
    ANY
    puts '11'
  end
end
```

Test the file
```sh
rubocop test.rb
```

## RuboCop version

```
[test_proj]$ rubocop -V
1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
```
